### PR TITLE
#61 Adding roles for different features

### PR DESCRIPTION
**Pull request by smaharj1**
PR for #61 
- Added roles for tariff and categories (VIEW/MANAGE)
- modified logic for tariff and categories route
- package version updates according to npm audit
- Rearranged the passportAuth middleware

### CODE REVIEW

File: management-server/routes/referenceData.js [hunk 1]
```diff
@@ -10,28 +10,28 @@ router.get('/', function(req, res, next) {
   res.render('index', { title: 'Veniqa Reference Data' });
 });
 
-router.get('/getCatalogBundle', passportAuth.isAuthenticated, referenceDataController.getCatalogBundle);
```
Comment (Viveckh) [inline management-server/routes/referenceData.js]: isAuthenticated checks if the user is authenticated. If we remove this, anyone can request it bbasically

File: management-server/routes/referenceData.js [hunk 2]
```diff
@@ -10,28 +10,28 @@ router.get('/', function(req, res, next) {
...
-router.get('/getStores', passportAuth.isAuthenticated, referenceDataController.getStores);
+router.get('/getStores', referenceDataController.getStores);
 
-router.get('/getRoles', passportAuth.isAuthenticated, referenceDataController.getRoles);
+router.get('/getRoles', referenceDataController.getRoles);
 
-router.get('/getWeightUnits', passportAuth.isAuthenticated, referenceDataController.getWeightUnits);
+router.get('/getWeightUnits', referenceDataController.getWeightUnits);
 
-router.get('/productCategoryList', passportAuth.isAuthenticated, passportAuth.canViewCatalog, referenceDataController.getProductCategoryList);
```
Comment (Viveckh) [inline management-server/routes/referenceData.js]: The canView functions don't check the authenticated status. So I think if we remove this, even if their session times out, the request will still be processed.

File: management-server/routes/referenceData.js (hunk 1 above)
Comment (smaharj1) [reply management-server/routes/referenceData.js]: its being checked in app.js

File: management-server/routes/referenceData.js (hunk 2 above)
Comment (smaharj1) [reply management-server/routes/referenceData.js]: isAuth is being checked in `app.js`. just removing multiple middlewares in one line

**Comment by smaharj1** [comment]
🔥

**Pull request merged**